presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -365,6 +366,13 @@ public PlanOptimizers(
                                 new TransformCorrelatedSingleRowSubqueryToProject())),
                 new CheckSubqueryNodesAreRewritten(),
                 predicatePushDown,
+                new IterativeOptimizer(
```

Comment:
Does this rule have to live above `TranslateExpressions`? If possible, we can have it between `TranslateExpressions` and `AddExchange`

Reply:
yep, intended to be above `TranslateExpressions`. Use OriginalExpressionUtils.

Reply:
But why? Any other specific rule it has to run after?

Reply:
ah...this work started long time ago, before `TranslateExpressions` introduced. When porting it to current code,  the implementation is based on `OriginalExpressionUtils` `cast`

Reply:
Can we have this patch written in `RowExpression` only? Check #14914